lib/core/src/server/build-static.js

```diff
@@ -193,12 +193,14 @@ export async function buildStaticStandalone(options) {
 
 export function buildStatic({ packageJson, ...loadOptions }) {
   const cliOptions = getProdCli(packageJson);
+  process.env.PREVIEW_URL = cliOptions.previewUrl;
 
   return buildStaticStandalone({
     ...cliOptions,
     ...loadOptions,
     packageJson,
     configDir: loadOptions.configDir || cliOptions.configDir || './.storybook',
     outputDir: loadOptions.outputDir || cliOptions.outputDir || './storybook-static',
+    ignorePreview: !!cliOptions.previewUrl,
```

Comment:
can we add `previewUrl` as another option here and pass it through to `loadEnv` below? I don't love messing with `process.env`

Reply:
I don't mind but I have no idea how the `Preview.js` component would be able to read form this "config" ? Do you have any example of a component reading from there ?

I initially thought that only the CLI /config tasks could read from the result of this function

Reply:
Oh, I think you misunderstand - I wasn't clear; I didn't mean don't set `process.env` for the webpack process, just don't mess with node's `process.env` on line 196 (☝️).

I'm pretty sure the `ignorePreview` option you set here will get passed through in the `options` argument of `loadEnv` below (in `utils.js` 👇) -- so if you also added `previewUrl` there, you could set the `PREVIEW_URL` env var there without needing to read it out of node's `process.env`.

Basically passing the option through rather than indirectly sending it via a global var (on `process.env`)